kubernetes/resource_kubernetes_ingress.go

```diff
@@ -167,63 +166,77 @@ func resourceKubernetesIngressCreate(d *schema.ResourceData, meta interface{}) e
 		}
 
 		if len(res.Status.LoadBalancer.Ingress) > 0 {
-			return resource.NonRetryableError(resourceKubernetesIngressRead(d, meta))
+			diagnostics := resourceKubernetesIngressRead(ctx, d, meta)
+			if diagnostics.HasError() {
+				errmsg := diagnostics[0].Summary
+				return resource.NonRetryableError(fmt.Errorf("Error reading ingress: %v", errmsg))
+			}
+			return nil
```

Comment:
Good catch! Returning `nil` instead of `diag.Diagnostics` caused a crash once when I was migrating the helm provider... I don't think the two can be used interchangeably like `error` and `nil` can.

Reply:
oh oops, this part actually is a function within the function. So returning `nil` is correct. (This function returns a `*resource.RetryError`).

Reply:
Oh yeah, totally missed that thanks!